### PR TITLE
use quotes instead of angle brackets to include armadillo header

### PR DESCRIPTION
**Pull request by conradsnicta**
Related to bug https://github.com/RcppCore/RcppArmadillo/issues/314

change `#include <armadillo>` to `#include "armadillo"`

### CODE REVIEW

**Comment by eddelbuettel** [comment]
As I just wrote "Maybe. Maybe not."  I'd prefer it we would not throw quick PRs around before issue tickets.

This has stood as is on mutiple OSs, versions, compilers, upgrades, for a _decade_.  Let's be sure that this is in fact it.

**Comment by conradsnicta** [comment]
> As I just wrote "Maybe. Maybe not." I'd prefer it we would not throw quick PRs around before issue tickets.
> 
> This has stood as is on mutiple OSs, versions, compilers, upgrades, for a _decade_. Let's be sure that this is in fact it.

It doesn't hurt changing `<...>` to `"..."`, and is likely to improve things.

Previous related issues: #281 #280 #264

**Comment by eddelbuettel** [comment]
Can we please STOP doubling and repeating all arguments in both blaces?

**Comment by eddelbuettel** [comment]
> It doesn't hurt changing <...> to "...", and is likely to improve things.

That's a conjecture.  It may, or it may not.

**Comment by conradsnicta** [comment]
ok, let's analyse this.

Background. A user has two installations of Armadillo:
- One installation is in `/usr/include` (or `/usr/local/include`)
- Another installation is wherever R stores its installed packages; as an example, let's say it's in `/home/user/R/x86_64-redhat-linux-gnu-library/3.6/RcppArmadillo/include/`

Having two installations is perfectly fine, as there are two clear use cases: (i) development of pure C++ code, and (ii) development of R mixed with C++ code.

Within the installed RcppArmadillo, we have the following file: `RcppArmadilloForward.h`.  This file is stored as `/home/user/R/x86_64-redhat-linux-gnu-library/3.6/RcppArmadillo/include/RcppArmadilloForward.h`.

Within `RcppArmadilloForward.h` we have the following line:  `#include <armadillo>`

The GCC docs on the include syntax (https://gcc.gnu.org/onlinedocs/cpp/Include-Syntax.html) mention two variants:

> `#include <file>`
> This variant is used for system header files. It searches for a file named file in a _standard list of system directories_. (...)
> 
> `#include "file"`
> This variant is used for header files of your own program. It searches for a file named file _first in the directory containing the current file_, then in the quote directories and then the same directories used for `<file>`. (...)

As `#include <armadillo>` is used within `RcppArmadilloForward.h`, the compiler will first search in standard directories like `/usr/include` and `/usr/local/include`. This will pick the system installation of Armadillo, and _ignore_ the installation in `/home/user/R/x86_64-redhat-linux-gnu-library/3.6/RcppArmadillo/include/`. We don't want this.

If `#include "armadillo"` is used instead within `RcppArmadilloForward.h`, the compiler will first search in the directory where `RcppArmadilloForward.h` is stored. In this example it will be `/home/user/R/x86_64-redhat-linux-gnu-library/3.6/RcppArmadillo/include/`. This is what we want.

File: ChangeLog [hunk 1]
```diff
@@ -1,3 +1,7 @@
+2020-10-24  Condrad Sanderson  <conradsand@gmail.com>
```
Comment (conradsnicta) [inline ChangeLog:1]: Please remove this email address.  If an email address is really necessary, I'm okay for the associated minor change in code to be attributed to to you, or perhaps "RcppArmadillo community" in general.

Comment (eddelbuettel) [reply ChangeLog:1]: No that's what we do. Changes get a ChangeLog entry.

**Comment by eddelbuettel** [comment]
The discussion at the gcc site is actually longer that what you quoted and also states that it should follow the include path as specified by the `-I` switches.

Which for some reason nobody can understand seems to be ordered the wrong way in the case of R on macOS.  

The issue is still bizarre because I happen to either maintain (or use) a number of other package with a similar structure of shipping their own header files which might well conflict with a system installation -- RcppEigen and RcppSpdlog are two good examples even if the latter is pretty recent.  And _e.g._ could have / should have thrown up the same issue but as far as I remember has not.  

Oh well, on balance this appears to be the right fix so I am going to merge this.

**Pull request merged**